### ISSUE TITLE
支持Arm64平台 / arm

### ISSUE THREAD

**Issue by sagehou** [chaitin/SafeLine#56]
如题，请问后续有无Arm64支持的计划~

**Comment by shangdev** [comment]
+1

**Comment by suruozhong** [comment]
+1，现在国产化系统，很多arm64的

**Comment by weihms** [comment]
目前大量企业面临国产化软件适配，需要兼容信创环境，ARM环境X86同时兼容，支持国产化转型，企业也能迎来发展机会，担当了社会使命。

**Comment by lihz1990** [comment]
+1

**Comment by zclaiqcc** [comment]
ARM、信创、国产化等需求目前判断都是企业级需求，有类似需求的师傅们真心建议去搞套企业版。

我个人判断，社区版短时间估计不会考虑这类支持。一方面是实现和维护的成本很高，光是 ARM 和国产化平台的适配的工作量已经非常非常大了，还不算未来各个组件的升级和维护。另一方面判断社区需求量不大，我个人不觉得社区用户会有类似信创这种需求。建议不要混淆企业需求和个人需求，社区版的目标不是替代企业版，不要有不切实际的幻想。

issue 继续 open，时机成熟的话也可以考虑支持

**Comment by zclaiqcc** [comment]
> 目前大量企业面临国产化软件适配，需要兼容信创环境，ARM环境X86同时兼容，支持国产化转型，企业也能迎来发展机会，担当了社会使命。

@weihms 大了大了，格局太大了。企业版早就支持了国产化信创，我们完成了该有的社会使命，就等师傅去买一套了

**Comment by boxdl** [comment]
+1

**Comment by dhfsqxz** [comment]
我们最近也有了国产化需求
不过社区版不上也行
毕竟有国产化需求的一般也不差这点钱

**Comment by pakchoily** [comment]
+1

**Comment by mozhou-tech** [comment]
+1 期待支持ARM64

**Comment by youlong1994** [comment]
+1 现在哪哪都是国产化 CPU国产 系统国产，急需Arm64架构

**Comment by xybingbing** [comment]
+1

**Comment by Mabisi** [comment]
+1

**Comment by hale177** [comment]
建议需要 arm  公司直接掏钱去买企业版，也算是支持 雷池社区 良性发展了

**Comment by cachexy123** [comment]
+1

**Comment by baozitao** [comment]
+1

**Comment by SunDaydream** [comment]
+1

**Comment by emiyadavid** [comment]
+1

**Comment by zcbxxx1** [comment]
+1

**Comment by emiyadavid** [comment]
好的，我已收到！

**Comment by cuicanmengxing** [comment]
+1

**Comment by Whbgz** [comment]
+1

**Comment by emiyadavid** [comment]
好的，我已收到！

**Comment by zclaiqcc** [comment]
> 好的，我已收到！

@emiyadavid 如果师傅继续定时回复无效内容，将会被拉黑

**Comment by binaryYuki** [comment]
社区版支持 arm64 架构 +1
目前似乎很多轻量服务器也是 arm 架构的 期待社区能支持

**Comment by emiyadavid** [comment]
好的，我已收到！

**Comment by Dullxxs** [comment]
+1，除了信创设备还有各种开发板.

**Comment by mxdwjcty** [comment]
we need arm64!!!

**Comment by szkzishan** [comment]
+1，希望能支持arm64

**Comment by emiyadavid** [comment]
好的，我已收到！

**Comment by chengzyzy** [comment]
+1

**Comment by vncloudsco** [comment]
+1

**Comment by zhaoyang1221** [comment]
希望能支持arm64架构

**Comment by cmodog** [comment]
顶，希望社区版可以支持arm64

**Comment by catiz** [comment]
同样希望社区版支持 arm64 架构，学生负担不起专业版

**Comment by Jerry12228** [comment]
+1
(我在想能不能自己构建容器镜像)
希望不要像宝塔一样, 越来越讨厌了, 商业化严重呐
拜托了

**Comment by wuuyun** [comment]
arm64 版waf 可以找其它平台的waf 
宝塔waf目前支持arm64

**Comment by Jerry12228** [comment]
> arm64 版waf 可以找其它平台的waf 宝塔waf目前支持arm64

您猜猜宝塔收费吗?

**Comment by wuuyun** [comment]
> > arm64 版waf 可以找其它平台的waf 宝塔waf目前支持arm64
> 
> 您猜猜宝塔收费吗?

免费的，已经在盒子上安装好了； 部分高级功能和雷池一样都是要收费的

**Comment by tianxingdao** [comment]
we need arm64!!!

**Comment by caiqy** [comment]
+1

**Comment by eniyiweb** [comment]
+1

**Comment by labulac** [comment]
+1

**Comment by eniyiweb** [comment]
In order for it to work on ARM devices, you need to purchase a business or professional account. I noticed this warning after I installed it.
![1](https://github.com/user-attachments/assets/6e2fd2d3-0d3a-40fe-938d-d45ddd3dfa7f)

**Comment by binaryYuki** [comment]
> In order for it to work on ARM devices, you need to purchase a business or professional account. I noticed this warning after I installed it. ![1](https://private-user-images.githubusercontent.com/51748469/393778444-6e2fd2d3-0d3a-40fe-938d-d45ddd3dfa7f.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3Mzc2OTksIm5iZiI6MTczMzczNzM5OSwicGF0aCI6Ii81MTc0ODQ2OS8zOTM3Nzg0NDQtNmUyZmQyZDMtMGQzYS00MGZlLTkzOGQtZDQ1ZGRkM2RmYTdmLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA5VDA5NDMxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJkNzkyYzM0ZWFjMmJkOWRlYTQwMTkzMWJjZDA0ZmI1MWYxYmFiMTczY2UyMzQ3NzBiNzZkMDQ3MDEzNjllMWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ucFrbB3qNOlLwBj4W3eWElnirVnuAeyvNIebuIRg24M)

Yes that's why we're here. As https://github.com/chaitin/SafeLine/issues/56#issuecomment-1535958815 says: "Community version is not available, the enterprise version is currently supported arm64, master can official website to consult the business version of the situation. Other masters can point a +1 if they have similar needs, and it's not impossible to support them in the community edition if they have a large demand 😊"

**Comment by eniyiweb** [comment]
> > In order for it to work on ARM devices, you need to purchase a business or professional account. I noticed this warning after I installed it. ![1](https://private-user-images.githubusercontent.com/51748469/393778444-6e2fd2d3-0d3a-40fe-938d-d45ddd3dfa7f.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3Mzc2OTksIm5iZiI6MTczMzczNzM5OSwicGF0aCI6Ii81MTc0ODQ2OS8zOTM3Nzg0NDQtNmUyZmQyZDMtMGQzYS00MGZlLTkzOGQtZDQ1ZGRkM2RmYTdmLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA5VDA5NDMxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJkNzkyYzM0ZWFjMmJkOWRlYTQwMTkzMWJjZDA0ZmI1MWYxYmFiMTczY2UyMzQ3NzBiNzZkMDQ3MDEzNjllMWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ucFrbB3qNOlLwBj4W3eWElnirVnuAeyvNIebuIRg24M)
> 
> Yes that's why we're here. As [#56 (comment)](https://github.com/chaitin/SafeLine/issues/56#issuecomment-1535958815) says: "Community version is not available, the enterprise version is currently supported arm64, master can official website to consult the business version of the situation. Other masters can point a +1 if they have similar needs, and it's not impossible to support them in the community edition if they have a large demand 😊"

Ok then +1 :)

**Comment by mifuo** [comment]
+1

**Comment by safe1ine** [comment]
专业版已经支持 arm 了

**Comment by rbernardes** [comment]
+1

**Comment by ZhongTerry** [comment]
+1
